contracts/schemes/GenericSchemeMultiCall.sol

```diff
@@ -0,0 +1,197 @@
+     * @param _contractWhitelist the contracts the scheme is allowed to interact with
+     * 
+     */
+    function initialize(
+        Avatar _avatar,
+        IntVoteInterface _votingMachine,
+        bytes32 _voteParams,
+        address[] calldata _contractWhitelist
+    )
+    external
+    {
+        require(avatar == Avatar(0), "can be called only one time");
+        require(_avatar != Avatar(0), "avatar cannot be zero");
+        require(_contractWhitelist.length > 0, "contractWhitelist cannot be empty");
+        avatar = _avatar;
+        votingMachine = _votingMachine;
+        voteParams = _voteParams;
+        for(uint i = 0; i < _contractWhitelist.length; i ++) {
+          contractWhitelist[_contractWhitelist[i]] = true;
+          whitelistedContracts.push(_contractWhitelist[i]);
+        }
+    }
+
+    /**
+    * @dev execution of proposals, can only be called by the voting machine in which the vote is held.
+    * @param _proposalId the ID of the voting in the voting machine
+    * @param _decision a parameter of the voting result, 1 yes and 2 is no.
+    * @return bool success
+    */
+    function executeProposal(bytes32 _proposalId, int256 _decision)
+    external
+    onlyVotingMachine(_proposalId)
+    returns(bool) {
+        MultiCallProposal storage proposal = proposals[_proposalId];
+        require(proposal.exist, "must be a live proposal");
+        require(proposal.passed == false, "cannot execute twice");
+
+        if (_decision == 1) {
+            proposal.passed = true;
+            execute(_proposalId);
+        } else {
+            delete proposals[_proposalId];
+            emit ProposalDeleted(address(avatar), _proposalId);
+        }
+
+        emit ProposalExecutedByVotingMachine(address(avatar), _proposalId, _decision);
+        return true;
+    }
+
+    /**
+    * @dev execution of proposals after it has been decided by the voting machine
+    * @param _proposalId the ID of the voting in the voting machine
+    */
+    function execute(bytes32 _proposalId) public {
+        MultiCallProposal storage proposal = proposals[_proposalId];
+        require(proposal.exist, "must be a live proposal");
+        require(proposal.passed, "proposal must passed by voting machine");
+        proposal.exist = false;
+        bytes memory genericCallReturnValue;
+        bool success;
+        Controller controller = Controller(avatar.owner());
+
+        for (uint i = 0; i < proposal.contractsToCall.length; i ++) {
+          if (proposal.contractsToCall[i] == address(controller)) {
+
+           (IERC20 extToken,
+            address spender,
+            uint256 valueToSpend
+           ) =
+           /* solhint-disable */
+           abi.decode(
+             proposal.callData[i],
+             (IERC20, address, uint256)
+           );
+           (success) = controller.externalTokenApproval(extToken,spender,valueToSpend,avatar);
+         } else {
+           (success, genericCallReturnValue) =
```

Comment:
if the first 4 bytes of the `genericCallReturnValue` are `0x08c379a0` that would mean that the call failed because the error message was throw but `success` can equals true, shouldnt we take that in count as call failed?

Reply:
reference: https://github.com/AugustoL/dxdao-contracts/blob/dxdao-contracts/contracts/schemes/WalletScheme.sol#L103

Reply:
Is there any documentation re this issue ?  I found this [old](https://github.com/ethereum/go-ethereum/issues/19027) one. 
Also , no reference for that in [OZ call](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/c801c8d2bbaeeab413fc792ae0d44f720fb81bc9/contracts/utils/Address.sol#L119) implementation.
@AugustoL is that reproducible ?

Reply:
@orenyodfat I think it is enough the way it is done by checking only the `bool success`, the signature of the data returned will be `0x08c379a0` in case there is a revert reason/message.